internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -242,6 +256,61 @@ func (r *BMACReconciler) reconcileAgentSpec(bmh *bmh_v1alpha1.BareMetalHost, age
 	return reconcileComplete{true}
 }
 
+// This function determines if the detached annotation should be added to the BMH
+// if the BMH is a SNO.
+//
+// The detached annotation is added if the installation of the agent associated with
+// the host has either started or has completed.
+func (r *BMACReconciler) updateBMHDetachedAnnotationIfSNO(ctx context.Context, bmh *bmh_v1alpha1.BareMetalHost, agent *aiv1beta1.Agent) reconcileResult {
+
+	bmhAnnotations := bmh.ObjectMeta.GetAnnotations()
+	// Annotation already set to true
+	if bmhAnnotations[BMH_DETACHED_ANNOTATION] == "true" {
+		return reconcileComplete{}
+	}
+
+	// No updates if agent does not have a ClusterDeployment name
+	if agent.Spec.ClusterDeploymentName == nil {
+		return reconcileComplete{}
+	}
```

Comment:
Maybe we can do away with this and move the agent conditions check here. What do you think?

this removes the need to check for a cluster deployment entirely (related to my comment below about setting detached regardless of the deployment setup)

Reply:
Yes. Sounds good.